src/certificate/commands/DefectsCommand.ts

```diff
@@ -0,0 +1,150 @@
+import { Service } from 'typedi';
+import { DefectRepository } from '../../defect/DefectRepository';
+import { DefectService } from '../../defect/DefectService';
+import { ITestResult } from '../../models';
+import { ICertificatePayload } from '../../models';
+import { ITestType } from '../../models';
+import { CERTIFICATE_DATA, TEST_RESULTS } from '../../models/Enums';
+import { IDefectParent } from '../../models/IDefectParent';
+import { IFlatDefect } from '../../models/IFlatDefect';
+import { BasePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class DefectsCommand extends BasePayloadCommand {
+	constructor(
+		private defectService: DefectService,
+		private defectRepository: DefectRepository
+	) {
+		super();
+	}
+
+	private certificateIsAnPassOrFail = (): boolean =>
+		this.state.type === CERTIFICATE_DATA.PASS_DATA || this.state.type === CERTIFICATE_DATA.FAIL_DATA;
+
+	public async generate(): Promise<ICertificatePayload> {
+		if (!this.certificateIsAnPassOrFail()) {
+			return {} as ICertificatePayload;
+		}
+
+		const {
+			testResult,
+			testResult: { testTypes },
+		} = this.state;
+
+		const result = {} as ICertificatePayload;
+
+		if (testTypes.testResult !== TEST_RESULTS.FAIL) {
+			result.DATA = {
+				...(await this.getPayloadData(testResult, CERTIFICATE_DATA.PASS_DATA)),
+			};
+		}
+
+		if (testTypes.testResult !== TEST_RESULTS.PASS) {
+			result.FAIL_DATA = {
+				...(await this.getPayloadData(testResult, CERTIFICATE_DATA.FAIL_DATA)),
+			};
+		}
```

Comment:
Just want to check, are the conditions done this way to include PRS? 

If not, it probably feels like we should use the positive assertion (if testResult is 'PASS', then use 'PASS_DATA') which to me reads better e.g.

```
if (testTypes.testResult === TEST_RESULTS.PASS) {
  result.DATA = {
    ...(await this.getPayloadData(testResult, CERTIFICATE_DATA.PASS_DATA)),
  };
}
```

Reply:
Yes I believe it was done this way to include PRS in the FAIL_DATA payload, the doc-gen logic looks for failData.prs

Reply:
Yeah I really don't like the conditions throughout cert gen. There's some real mental gymnastics required to parse it all.

In later PRs we really need to refactor these to be more intentional.